### PR TITLE
Feature/sof 5954 2

### PR DESCRIPTION
**Pull request by azech-hqs**

### CODE REVIEW

File: schema/method_units/mathematical/opt/diff/ordern.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "schemaId": "method-unit-opt-diff-ordern",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "description": "Mixed order and higher order algorithms for the optimization of differentiable functions",
```
Comment (azech-hqs) [inline schema/method_units/mathematical/opt/diff/ordern.json:4]: I had to create a new tier for mixed or higher order optimization methods as conjugate gradient is often described as order 1.5. See here for instance: https://optimization.cbe.cornell.edu/index.php?title=Conjugate_gradient_methods

**Review by tjduigna — APPROVED** [review]

File: schema/method_units/physical/rmm_diis.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+    "schemaId": "method-derived-qm-rmm-diis",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "Residual minimization method using direct inversion of the iterative subspace (RMM-DIIS) schema",
+    "description": "",
```
Comment (timurbazhirov) [inline schema/method_units/physical/rmm_diis.json]: Maybe let's add a DOI link to the article?

File: schema/method_units/physical/rmm_diis.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            ]
+        },
+        "parameters": {
+            "type": "object",
+            "properties": {
+                "blockSize": {
+                    "description": "Number of eigenvalues to be solved simultaneously (blocked mode)",
+                    "$ref": "../../core/reusable/parameters/int.json"
+                },
+                "diis.maxSteps": {
```
Comment (timurbazhirov) [inline schema/method_units/physical/rmm_diis.json]: I'd suggest spelling out the keys

File: example/methods_directory/cg_diag.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        },
+        {
+            "flowchartId": "um00002",
+            "methodPath": "mathematical/linalg/diag/davidson",
+            "tier1": {
+                "name": "Linear Algebra",
+                "slug": "linalg"
+            },
+            "parameters": {
+                "nEigval": {
```
Comment (timurbazhirov) [inline example/methods_directory/cg_diag.json]: Should be full names, I guess, like in https://github.com/Exabyte-io/esse/pull/210/files#diff-d13e8de21146d779fbf6293b238b528a854bff3abc1278c8bda90868c3a54fc0R21

**Pull request merged**